tools/pkg/engine_build_configs/test/ci_yaml_test.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+
+import 'package:engine_build_configs/src/ci_yaml.dart';
+import 'package:engine_repo_tools/engine_repo_tools.dart';
+import 'package:litetest/litetest.dart';
+import 'package:path/path.dart' as path;
+import 'package:source_span/source_span.dart';
+import 'package:yaml/yaml.dart' as y;
+
+void main() {
```

Comment:
Can we have a test to validate the logic when other unrelated configs exist? What if there exists a malformed config there? Do we want to ignore them whenever the desired ones are valid?

Reply:
I think if any config looks invalid, then it's okay for the whole parse to fail. The whoel input .ci.yaml has to be valid since it would fail in presubmit if it's invalid, anyway. I added another test with both a valid and invalid target to check that behavior.